integration/util.go

```diff
@@ -200,30 +219,36 @@ func (k *NSKubernetesClient) WaitForPodsInPhase(expectedPhase v1.PodPhase, podNa
 			//k.debug("nodes")
 			k.debug("pods")
 			k.logs("pod", podNames)
-			k.t.Fatalf("Timed out waiting for pods %v ready in namespace %s", podNames, k.ns)
+			k.t.Fatalf("Timed out waiting for pods %v in namespace %q", podNames, k.ns)
 
 		case event := <-w.ResultChan():
 			if event.Object == nil {
 				return
 			}
 			pod := event.Object.(*v1.Pod)
-			logrus.Infoln("Pod", pod.Name, "is", pod.Status.Phase)
 			if pod.Status.Phase == v1.PodFailed {
 				logs, err := pods.GetLogs(pod.Name, &v1.PodLogOptions{}).DoRaw(ctx)
 				if err != nil {
 					k.t.Fatalf("failed to get logs for failed pod %s: %s", pod.Name, err)
 				}
 				k.t.Fatalf("pod %s failed. Logs:\n %s", pod.Name, logs)
 			}
-			phases[pod.Name] = pod.Status.Phase
 
+			if _, found := podsReady[pod.Name]; !found && waitForAllPods {
+				podNames = append(podNames, pod.Name)
+			}
+			podsReady[pod.Name] = podReady(pod)
+
+			var waiting []string
 			for _, podName := range podNames {
```

Comment:
would this cause any issues when waitForAllPods is true? I'm just thinking that maybe a pod goes through all the necessary phases before any other and so it ends up being the only one in `podNames` resulting in `waiting` being empty despite there potentially being more pods who haven't reached a phase yet.

Reply:
That is true: there is a race condition where pod creation events are pending in the watcher.  I guess I could check `len(w.ResultChan())` to see if there are other events pending.

Reply:
I hope these pods will eventually reach a fixpoint :-/  I suppose we do have a timeout.  I print out how many pending messages there are.